superdesk/users/services.py

```diff
@@ -411,3 +413,62 @@ def on_deleted(self, doc):
 
         super().on_deleted(doc)
         get_resource_service('reset_user_password').remove_all_tokens_for_email(doc.get('email'))
+
+
+class UserSessionClearResource(Resource):
+    endpoint_name = 'clear_sessions'
+    url = 'users/<{0}:user_id>/clear_sessions'.format(item_url)
+    datasource = {'source': 'users'}
+    resource_methods = ['POST']
+    resource_title = endpoint_name
+    privileges = {'POST': 'users'}
+
+
+class UserSessionClearService(BaseService):
+
+    def create(self, docs, **kwargs):
+        """
+        Deletes all the records from auth and corresponding
+        session_preferences from user collections
+        If there are any orphan session_preferences exist they get deleted as well
+        """
+        users_service = get_resource_service('users')
+        user_id = request.view_args['user_id']
+        user = users_service.find_one(req=None, _id=user_id)
+        sessions = get_resource_service('auth').get(req=None, lookup={'user': user_id})
+
+        error_message = self.__can_clear_sessions(user)
+        if error_message:
+            raise SuperdeskApiError.forbiddenError(message=error_message)
+
+        # Delete all the sessions
+        for session in sessions:
+            get_resource_service('auth').delete_action({config.ID_FIELD: str(session[config.ID_FIELD])})
+
+        # Check if any orphan session_preferences exist for the user
+        if user.get('session_preferences'):
+            # Delete the orphan sessions
+            # users_service.system_update(user_id, {'first_name': 'hebe'}, user)
+            users_service.patch(user[config.ID_FIELD], {'session_preferences': {}})
+
+        return [{'complete': True}]
+
+    def __can_clear_sessions(self, user):
+        """
+        Checks if the session clear request is Invalid.
+        Operation is invalid if one of the below is True:
+            1. Check if the user exists.
+            2. Check if the user is clearing his/her own sessions.
+            3. A user without 'User Management' privilege is clearing the sessions
+
+        :return: error message if invalid.
+        """
+
+        if not user:
+            return 'Invalid user to clear sessions.'
+
+        if not current_user_has_privilege('users'):
```

Comment:
I think this is redundant as `SuperdeskAuth.check_privileges()` takes care of this